### ISSUE TITLE
Mismatch mounted hook explanation in "Migration from Vue 1.x" vs. "API v2"

### ISSUE THREAD

**Issue by nolimitdev** [vuejs/vuejs.org#1922]
From https://vuejs.org/v2/guide/migration.html#ready-replaced

> It should be noted though that with mounted, there’s no guarantee to be in-document.

From https://vuejs.org/v2/api/#mounted

> Note that mounted does not guarantee that all child components have also been mounted. If you want to wait until the entire view has been rendered, you can use vm.$nextTick inside of mounted

Text which explains mounted hook in migration is different from the text in API v2. Text in migration should be updated to be the same such as in API v2 because term `no guarantee` is not explained. Why? Mounted hook guarantee that component is in-document when its parent is in-document. So when component does not have children it is useless to always use $nextTick.

**Comment by phanan** [comment]
I'm not sure I get what you mean. You're saying "Text which explains mounted hook in migration is different from the text in API v2" but they are the same:

> there’s _no_ guarantee to be in-document.

and 

> does _not_ guarantee that all child components have also been mounted

What am I missing here?

**Comment by nolimitdev** [comment]
Yes, they are different. Try to read my post again :) Note word "ALSO" in your second quotation... it means that current vue instance/component on which was mounted hook called HAVE TO BE mounted (it is guaranteed) but only children DO NOT HAVE TO BE mounted (it is not guaranteed).

So https://vuejs.org/v2/api/#mounted correctly says that only children do not have to be mounted.

And migration guide https://vuejs.org/v2/guide/migration.html#ready-replaced says that current vue instance/component do not have to be mounted so it creates a false impression that we always must use $nextTick also if code in mounted hook only wants to access DOM element (e.g via document.getElementById('...')) which is a part of its (hooks`s) instance/component.

**Comment by nolimitdev** [comment]
https://stackoverflow.com/questions/54482247/vue-js-note-that-mounted-does-not-guarantee-that-all-child-components-have-also
